### DISCUSSION TITLE
error - OperationProcessingError: "response" is not a conform Authorization Server Metadata response

### DISCUSSION THREAD

**Discussion by paulwongx** [panva/oauth4webapi#68] [Q&A]
Super new to this library and not an expert in oauth flow but... based on the [examples for Authorization Code Flow](https://github.com/panva/oauth4webapi/blob/main/examples/oauth.ts) using discoveryRequest... I'm getting errors at the very first part.

If I try with Google and `oidc`:
```js
const issuer = new URL("https://accounts.google.com");
const discoveryResponse = await oauth.discoveryRequest(issuer, { algorithm: 'oidc' })
const as = await oauth.processDiscoveryResponse(issuer, discoveryResponse);

// Getting the error
// error - TypeError [ERR_INVALID_URL]: Invalid URL
```

So I think the discoveryResponse is not working... when I try to console log, I'm getting...
```js
const issuer = new URL("https://github.com/login/oauth/authorize");
const discoveryResponse = await oauth.discoveryRequest(issuer, { algorithm: 'oauth2' })
console.log({discoveryResponse: await discoveryResponse.json()});
// { discoveryResponse: { error: 'Not Found' } }
```

So the url is wrong...? Am I misunderstanding how this should work or...?

**Edit:** It works when I copy the `discoveryRequest` function into my own code and call it. I guess there's something really funky going on in my end, but not sure what...

**Comment by paulwongx** [comment]
For some reason I'm getting a `404` response but looking at the library code, for Google for example, it should be hitting this route `https://accounts.google.com/.well-known/openid-configuration` which works when I go to it directly, but for some reason it doesn't work when I call it via
```js
const issuer = new URL("https://accounts.google.com");
const discoveryResponse = await oauth.discoveryRequest(issuer)
``` 
When I call it manually, it works...
```js
  const res = await fetch("https://accounts.google.com/.well-known/openid-configuration",{
    headers: {
      "accept": "application/json"
    },
    method: "GET",
    redirect: "manual",
  })

  console.log(await res.json());
```

**Reply by paulwongx** [reply]
Hm, this is very confusing... when I copy the `discoveryRequest` code into my own code it works. When I console log the `discoveryRequest` function in my node_modules, it doesn't console log anything. I guess there's something really weird going on in my end.... but no idea what. I'm using the latest v2.3.0 too...

**Comment by panva** [comment]
```
const issuer = new URL("https://accounts.google.com");
const discoveryResponse = await oauth.discoveryRequest(issuer, { algorithm: 'oidc' })
const as = await oauth.processDiscoveryResponse(issuer, discoveryResponse);
```

This is correct.

```
// Getting the error
// error - TypeError [ERR_INVALID_URL]: Invalid URL
```

I'm not. I assume something is amiss in your runtime.

**Reply by paulwongx** [reply]
Thanks. Yeah it's something on my end. Appreciate your response!
